sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -0,0 +1,1050 @@
+= SYCL_EXT_ONEAPI_GRAPH
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:sectnums:
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+== Notice
+
+Copyright (c) 2022 Intel Corporation.  All rights reserved.
+
+IMPORTANT: This specification is a draft.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are
+trademarks of The Khronos Group Inc. OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+
+This extension is written against the SYCL 2020 revision 5 specification. All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+== Contributors
+
+Pablo Reble, Intel +
+Julian Miller, Intel +
+John Pennycook, Intel +
+Guo Yejun, Intel +
+Ewan Crawford, Codeplay +
+Ben Tracy, Codeplay +
+Duncan McBain, Codeplay +
+Peter Žužek, Codeplay +
+Ruyman Reyes, Codeplay +
+Gordon Brown, Codeplay +
+Erik Tomusk, Codeplay +
+Bjoern Knafla, Codeplay +
+Lukas Sommer, Codeplay +
+Ronan Keryell, AMD +
+
+== Status
+
+This is a proposed extension specification, intended to gather community
+feedback.  Interfaces defined in this specification may not be implemented yet
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+== Introduction
+
+Through the use of command groups SYCL is already able to create a dependency
+graph (in the form of a directed acyclic graph) of kernel execution at runtime,
+as a command group object defines a set of requisites (edges) which must be
+satisfied for kernels (nodes) to be executed. However, because command-group
+submission is tied to execution on the queue, without having a prior
+construction step before starting execution, optimization opportunities are
+missed from the runtime not knowing the complete dependency graph ahead of
+execution.
+
+The following benefits would become possible if the user could define a
+dependency graph to the SYCL runtime prior to execution:
+
+* Reduction in runtime overhead by only submitting a single graph object, rather
+  than many individual commands.
```

Comment:
In this case I believe we are referring to command groups since we are talking about submissions to a queue rather than commands with a command group. Will update the text to be more clear!